dune/stuff/functions/expression.hh

```diff
@@ -242,6 +250,35 @@ public:
     build_gradients(variable, gradient_expressions);
   }
 
+  // constructors taking a FieldMatrix for the jacobian
+  template< int rows, int cols >
+  Expression(const std::string variable,
+             const std::string expression,
+             const size_t ord = default_config().get< size_t >("order"),
+             const std::string nm = static_id(),
+             const Dune::FieldMatrix< std::string, rows, cols > jacobian
+                = Dune::FieldMatrix< std::string, 0, 0 >())
+    : function_(new MathExpressionFunctionType(variable, expression))
+    , order_(ord)
+    , name_(nm)
+  {
+    build_gradients(variable, jacobian);
+  }
+
+  template< int rows, int cols >
+  Expression(const std::string variable,
+             const std::vector< std::string > expressions,
+             const size_t ord = default_config().get< size_t >("order"),
+             const std::string nm = static_id(),
+             const Dune::FieldMatrix< std::string, rows, cols > jacobian
+                = Dune::FieldMatrix< std::string, 0, 0 >())
```

Comment:
I think it did compile... I will check again. Shouldn't the compiler be able to figure out that rows = cols = 0 if the default value is used?